### DISCUSSION TITLE
Current Allennlp Docker Image - has the userfunction as super user.

### DISCUSSION THREAD

**Discussion by mshanmugam** [allenai/allennlp#4980] [Q&A]
Hi Team ,

Current Allenai docker image is having super user(sudo/as root)  function which we cannot bring it inside our own environments. Is there a way we can we can disable them via any configurations / env variables.

**Comment by epwalsh** [comment]
Hey @mshanmugam, I'm not 100% sure what you mean. Are you talking about how you have to run Docker with sudo permissions?

**Comment by dirkgr** [comment]
Are you trying to run the AllenNLP docker image on a computer where you can't run Docker? An alternative way to go might be to install [miniconda](https://docs.conda.io/en/latest/miniconda.html), and then install AllenNLP into the conda environment (with `pip install allennlp`). You'll have to learn a bit about conda though.

**Reply by mshanmugam** [reply]
Actually we are trying to use the docker image as is and our security scan says , allennlp docker image has some root privilege function so we couldn't pass the scan. 

On the other note , we were able to manually install the allennlp on docker kubernetes platform , with 2 core cpu to the container and 4GB  memory , we are seeing 10X slower performance on container compared to the baremetal/regular VM. 
we don't use GPU , we use only CPU(No Cuda/GPU)  on both containers and baremetal. The model that is being used is BiDAF with glove embedding for reading comprehension. Please let me know if  need to open a separate discussion thread for the slowness / performance. 

Thanks for your help.

**Reply by dirkgr** [reply]
Does it give any extra information? What root privilege function is it complaining about?

I would guess the performance issue is down to the version of torch you're installing. Is there a difference between these versions?

**Reply by mshanmugam** [reply]
For Root ,it doesnt say any specific commands , it just says the image needs to function as a super user. 

For the performance the torch libs are same .. We are using 2.1.0 model , and torch version is 1.7.1 on python 3.6.  
Only difference is baremetal/vm has 8 cores , the container is allocated with 2 cores.  But I couldn't visibly see the cpu usage being high on containers,its using less than 100 milli cores.

**Reply by dirkgr** [reply]
There are different ways torch 1.7.1 could be built, even when it's just doing CPU. I tend to use the version from [Conda](https://docs.conda.io/en/latest/) all the time, but if you `pip install`, you will get a different version.
